x/evm/types/codec.go

```diff
@@ -9,14 +9,13 @@ var ModuleCdc = codec.New()
 
 func init() {
 	cdc := codec.New()
-
 	codec.RegisterCrypto(cdc)
-
 	ModuleCdc = cdc.Seal()
 }
 
 // RegisterCodec registers concrete types and interfaces on the given codec.
 func RegisterCodec(cdc *codec.Codec) {
-	cdc.RegisterConcrete(&EthereumTxMsg{}, "ethermint/MsgEthereumTx", nil)
-	cdc.RegisterConcrete(&EmintMsg{}, "ethermint/MsgEmint", nil)
+	cdc.RegisterConcrete(&MsgEthereumTx{}, "ethermint/MsgEthereumTx", nil)
+	cdc.RegisterConcrete(&MsgEthermint{}, "ethermint/MsgEmint", nil)
+	cdc.RegisterConcrete(EncodableTxData{}, "ethermint/EncodedMessage", nil)
```

Comment:
I actually don't think this type is needed, I remember there was a reason before. Can probably remove that type altogether, but I can check in more detail later